File: leet-code/leet-code/src/main/java/org/example/arrays/TwoSum.java
package org.example.arrays;

import java.util.HashMap;
import java.util.Map;

public class TwoSum {

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> secondInt = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (secondInt.containsKey(nums[i])) {
                return new int[]{secondInt.get(nums[i]), i};
            }
            secondInt.put(target - nums[i], i);
        }
        return new int[]{};
    }
}

File: first-kotlin/src/main/kotlin/leetcode/TwoSum.kt
package leetcode

fun main() {
    val nums = intArrayOf(3, 4, 20, 100)

    println(twoSum(nums, 103).joinToString())
}

/**
 * Given an array of integers nums and an integer target,
 * return indices of the two numbers such that they add up to target.
 *
 * You may assume that each input would have exactly one solution,
 * and you may not use the same element twice.
 *
 * You can return the answer in any order.
 */
fun twoSum(nums: IntArray, target: Int): IntArray {
    val secondInt: MutableMap<Int, Int> = HashMap()
    for (i in nums.indices) {
        if (secondInt.containsKey(nums[i])) {
            return intArrayOf(secondInt[nums[i]]!!, i)
        }
        secondInt[target - nums[i]] = i
    }
    return intArrayOf()
}